### PR TITLE
fix(17857): show correctly converted account balance in sign & encrypt 

### LINKED ISSUE #17857: [Bug]: Incorrect balance shown in the Request encryption public key confirmation screen header
### Describe the bug

Request encryption public key header balance

The USD value is shown in the header, instead of the ETH value

Related to: #17541

### Steps to reproduce

1. Open MM
2. On a new browser tab, go to Test Dapp
3. Connect to Test Dapp
4. Click Get Encryption Keys

Primary Currency: GoerliETH
Currency Conversion: USD
Show conversion on test networks: ON
Show balance and token price checker: ON

### Error messages or log output

_No response_

### Version

10.26

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DDDDDanica**
Fix: https://github.com/MetaMask/metamask-extension/issues/17857

## Explanation

This bug is originally from https://github.com/MetaMask/metamask-extension/pull/17334.

In the original implementation, it seems like we are using a value `conversionRate`
```
    const nativeCurrencyBalance = conversionUtil(balance, {
      fromNumericBase: 'hex',
      toNumericBase: 'dec',
      fromDenomination: 'WEI',
      numberOfDecimals: 6,
      conversionRate,
    });
```

however, `conversionRate` is only quoted when we convert between different currencies, and in our case, we are not passing `fromCurrency` or `toCurrency`
```
  if (fromCurrency !== toCurrency) {
    if (conversionRate === null || conversionRate === undefined) {
      throw new Error(
        `Converting from ${fromCurrency} to ${toCurrency} requires a conversionRate, but one was not provided`,
      );
    }
    let rate = toBigNumber.dec(conversionRate);
    if (invertConversionRate) {
      rate = new BigNumber(1.0).div(conversionRate);
    }
    convertedValue = convertedValue.times(rate);
  }
```

Hence, it is unnecessary to pass `conversionRate` in the new `Numeric` module.

Moreover,   `.toDenomination(EtherDenomination.ETH)` is also necessary for `Numeric` module to walk inside `denomination && this.denomination !== denomination` condition and add `toSpecifiedDenomination` to final value.

This incorrect transformation is occurring in 3 places, `Sign Typed Data V3`, `Get Encryption Key` as well as `Decrypt message`. If you encounter any other places displaying wrongly, feel free to comment in this MR so we can collect them together.

In which cases, for `Sign Typed Data V3`, a unit test snapshot is updated to reflected the correct value, where for other 2 features an extra assertion is added in related e2e tests.

### Before

See https://github.com/MetaMask/metamask-extension/issues/17857

### After
Screenshot 2023-03-02 at 22 51 37
Screenshot 2023-03-02 at 22 55 18

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: test/data/mock-state.json [hunk 1]
```diff
@@ -193,7 +193,7 @@
     "selectedAddress": "0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc",
     "accounts": {
       "0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc": {
-        "balance": "0x0",
+        "balance": "0x346ba7725f412cbfdb",
```
Comment (DDDDDanica) [inline test/data/mock-state.json:196]: This change is to validate the balance in`Sign typed data v3` feature

File: ui/components/app/signature-request-original/__snapshots__/signature-request-original.test.js.snap [hunk 2]
```diff
@@ -167,7 +167,7 @@ exports[`SignatureRequestOriginal should match snapshot 1`] = `
           <h6
             class="box box--margin-bottom-1 box--flex-direction-row typography typography--h6 typography--weight-bold typography--style-normal typography--align-end typography--color-text-default"
           >
-            0
+            966.987986
```
Comment (DDDDDanica) [inline ui/components/app/signature-request-original/__snapshots__/signature-request-original.test.js.snap:170]: This change is expected due to a mockdata change and is to validate the balance inSign typed data v3 feature

**Review by danjm — APPROVED** [review]
great work!

**Review by pedronfigueiredo — APPROVED** [review]

**Comment by DDDDDanica** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**